src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -77,7 +77,11 @@
     URI location = pathAsURI(job.get(TABLE_LOCATION));
 
     // Set defaults for virtual column
-    currentSnapshotId = table.currentSnapshot().snapshotId();
+    try {
+      currentSnapshotId = table.currentSnapshot().snapshotId();
+    } catch (NullPointerException e) {
+      LOG.warn("No snapshot available for table - table is empty.");
```

Comment:
If we have the null check we probably don't need to log this, querying an empty table is a "normal" thing and will just return no results right?

Reply:
Yeah true, and when testing this thats what happens. I didn't want to leave the catch empty, and wasn't sure of another way of not running into the NPE